### PR TITLE
Reorganize sample pipelines in "cache-enabled" and "cache-disabled" folders

### PR DESCRIPTION
**Pull request by jgarciao**
- Reorganizes sample pipelines in "cache-enabled" and "cache-disabled" folders
- Python files reformatted with `ruff format`. Linter warnings fixed or disabled

Signed-off-by: Jorge Garcia Oncins <jgarciao@redhat.com>

### CODE REVIEW

**Review by FedeAlonso — DISMISSED** [review]

**Comment by jgarciao** [comment]
Tested successfully in rhods-ci-pr-test/3382  (1 test failing for unrelated reasons)

**Review by FedeAlonso — APPROVED** [review]

File: ods_ci/libs/DataSciencePipelinesKfp.py [hunk 1]
```diff
@@ -459,7 +459,9 @@ def create_run_from_pipeline_func(
         # the current_path will be ods-ci
         if current_path is None:
             current_path = os.getcwd()
-        my_source = self.import_souce_code(f"{current_path}/tests/Resources/Files/pipeline-samples/v2/{source_code}")
+        my_source = self.import_souce_code(
+            f"{current_path}/tests/Resources/Files/pipeline-samples/v2/cache-disabled/{source_code}"
```
Comment (diegolovison) [inline ods_ci/libs/DataSciencePipelinesKfp.py:463]: The default for KFP is `cache-enabled`. We should test the defaults first. Disabling the cache can be a parameter.

File: ods_ci/tests/Tests/1100__data_science_pipelines/1105__data-science-pipelines-gpu-nvidia.robot [hunk 2]
```diff
@@ -11,7 +11,7 @@ Suite Teardown      Dsp Nvidia Gpu Suite Teardown
 *** Variables ***
 # robocop: off=line-too-long
 ${PROJECT}=    dsp-gpu-nvidia
-${PIPELINE_GPU_AVAILABILITY_FILEPATH}=    tests/Resources/Files/pipeline-samples/v2/gpu/pytorch/pytorch_verify_gpu_availability_compiled.yaml
+${PIPELINE_GPU_AVAILABILITY_FILEPATH}=    tests/Resources/Files/pipeline-samples/v2/cache-disabled/gpu/pytorch/pytorch_verify_gpu_availability_compiled.yaml
```
Comment (diegolovison) [inline ods_ci/tests/Tests/1100__data_science_pipelines/1105__data-science-pipelines-gpu-nvidia.robot:14]: Same for robot files

**Review by diegolovison — CHANGES_REQUESTED** [review]
The default setting in KFP is to have caching enabled. Changing the default in ODS-CI to have caching disabled makes it more difficult to remember the original default behavior. We should align with the upstream configuration.

File: ods_ci/libs/DataSciencePipelinesKfp.py (hunk 1 above)
Comment (jgarciao) [reply ods_ci/libs/DataSciencePipelinesKfp.py]: I've addressed this comment as per our discussion in slack

**Review by diegolovison — APPROVED** [review]

**Review by FedeAlonso — APPROVED** [review]

File: ods_ci/tests/Tests/1100__data_science_pipelines/1105__data-science-pipelines-gpu-nvidia.robot (hunk 2 above)
Comment (jgarciao) [reply ods_ci/tests/Tests/1100__data_science_pipelines/1105__data-science-pipelines-gpu-nvidia.robot:14]: discussed on slack

**Pull request merged**